### PR TITLE
MNT: Remove doc build from CircleCI (use RTD PR builder)

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to remove documentation build job from CircleCI and rely on [RTD PR builder](https://docs.readthedocs.io/en/latest/guides/autobuild-docs-for-pull-requests.html). The feature is already enabled on the admin side, but let's see if it really works. 🤞 

rtd_astropy

### CODE REVIEW

**Comment by Cadair** [comment]
Something that has come up in the discussions around this with SunPy is RTD doesn't catch sphinx warnings. I don't know of anyway to make it either...

**Comment by pllim** [comment]
Actually, it can. Please see second commit (and thanks for reminding me to do it).

**Comment by Cadair** [comment]
I guess the drawback of that is it's going to stop the actual deploy builds from passing in those cases as well, but as we are building them on all the PRs that's less of an issue.

**Comment by pllim** [comment]
You are correct. I think the pros outweighs the cons here. If failing on warning during deploy is a show-stopper, we can perhaps submit a feature request to RTD so that directive can be ignored for deploy event (although I don't think it is necessary).

File: tox.ini [hunk 1]
```diff
@@ -1,7 +1,6 @@
 [tox]
 envlist =
     py{36,37,38,dev}-test{,-alldeps,-oldestdeps,-devdeps,-numpy116,-numpy117,-numpy118}{,-cov}
-    build_docs
```
Comment (Cadair) [inline tox.ini:4]: We shouldn't remove this.

File: tox.ini [hunk 2]
```diff
@@ -93,14 +92,6 @@ commands =
     cov-!double: pytest --pyargs astropy {toxinidir}/docs {env:MPLFLAGS} --cov astropy --cov-config={toxinidir}/setup.cfg {posargs}
     double: python -c 'import sys; from astropy import test; test(); sys.exit(test())'
 
-[testenv:build_docs]
-changedir = docs
-description = invoke sphinx-build to build the HTML docs
-extras = docs
-commands =
-    pip freeze
-    sphinx-build -W -b html . _build/html
```
Comment (Cadair) [inline tox.ini:102]: We should keep this as well.

I am +1 on removing the `-W` here to fix issues with building docs on mac, but I suggest we actually punt that discussion to another PR / issue.

Comment (pllim) [reply tox.ini:102]: Ah, okay. So we keep this so that developers can use tox to build docs but *not* use it in CI, correct?

Comment (pllim) [reply tox.ini:102]: Anyways, I reverted the deletion as you requested.

**Comment by pllim** [comment]
Given an approval is given, I am just going to merge this now because new PRs now have duplicate doc builds, which is quite unnecessary. 😬 

Thanks, @Cadair !

**Pull request merged**